components/tables/editable-row.js

```diff
@@ -8,98 +18,144 @@ const EditableRow = ({
+		to_account: toAccount,
+		amount: toDollars(row.amount),
+		from_balance: toDollars(row.from_balance),
+		to_balance: toDollars(row.to_balance),
+		comment: row.comment
+	});
 
 	const handleSave = () => {
-		save(editedRow, originalRow, index);
+		const newDate = editedRow.trn_id
+			.toISOString()
+			.replace(/^(\d{4})-(\d{2})-(\d{2}).+/, '$1$2$3');
+
+		const editedRowInCents = {
+			trn_id: newDate,
+			from_account: editedRow.from_account.value,
+			to_account: editedRow.to_account.value,
+			amount: toCents(editedRow.amount.replace(/[$,]/g, '')),
+			from_balance: toCents(editedRow.from_balance.replace(/[$,]/g, '')),
+			to_balance: toCents(editedRow.to_balance.replace(/[$,]/g, '')),
+			comment: editedRow.comment
+		};
+
+		save(editedRowInCents, originalRow);
 	};
 
 	const handleCancel = () => {
-		setEditedRow(originalRow);
 		cancel(originalRow);
 	};
 
 	return isEditing && idBeingEdited === row.trn_id ? (
 		<tr>
 			<td>
-				<input
-					value={editedRow.trn_id}
-					onChange={() => handleEdit('trn_id', event)}
-				/>
+				<DateSelector setEditedRow={setEditedRow} editedRow={editedRow} />
 			</td>
 			<td>
-				<input
-					value={editedRow.from_account}
-					onChange={() => handleEdit('from_account', event)}
+				<FromAccountSelector
+					accounts={accounts}
+					setEditedRow={setEditedRow}
+					editedRow={editedRow}
 				/>
 			</td>
 			<td>
-				<input
-					value={editedRow.to_account}
-					onChange={() => handleEdit('to_account', event)}
+				<ToAccountSelector
+					accounts={accounts}
+					setEditedRow={setEditedRow}
+					editedRow={editedRow}
 				/>
 			</td>
 			<td>
-				<input
-					value={editedRow.amount}
-					onChange={() => handleEdit('amount', event)}
-				/>
+				<AmountSelector value={editedRow.amount} setEditedRow={setEditedRow} />
 			</td>
+			{showBalances && (
+				<>
+					<td>
+						<FromBalanceSelector
+							setEditedRow={setEditedRow}
+							value={editedRow.from_balance}
+						/>
+					</td>
+					<td>
+						<ToBalanceSelector
+							setEditedRow={setEditedRow}
+							value={editedRow.to_balance}
+						/>
+					</td>
+				</>
+			)}
 			<td>
-				<input
-					value={editedRow.from_balance}
-					onChange={() => handleEdit('from_balance', event)}
-				/>
-			</td>
-			<td>
-				<input
-					value={editedRow.to_balance}
-					onChange={() => handleEdit('to_balance', event)}
-				/>
-			</td>
-			<td>
-				<input
+				<CommentSelector
+					setEditedRow={setEditedRow}
 					value={editedRow.comment}
-					onChange={() => handleEdit('comment', event)}
 				/>
 			</td>
 
 			<td>
-				<button type={'button'} onClick={handleCancel}>
-					Cancel
-				</button>
 				<button type={'button'} onClick={handleSave}>
-					Save
+					{'Save'}
+				</button>
+				<button type={'button'} onClick={handleCancel}>
+					{'Cancel'}
 				</button>
 			</td>
 		</tr>
 	) : (
 		<tr>
-			{Object.values(row).map(cell => (
-				<td key={cell}>{cell}</td>
-			))}
+			<td key={`${index}-id`}>
```

Comment:
Row being edited vs not edited should be in separate components. This one is getting too large